lib/src/commit_builder.rs

```diff
@@ -332,7 +332,9 @@ impl DetachedCommitBuilder {
     }
 
     pub fn set_sign_key(&mut self, sign_key: Option<String>) -> &mut Self {
-        self.sign_settings.key = sign_key;
+        if sign_key.is_some() {
+            self.sign_settings.key = sign_key;
+        }
         self
     }
```

Comment:
nit: looking at the overall method now (my previous nit was a mechanical refactor to avoid an unnecessary clone) - `set_sign_key` does nothing if given option is None, which is a bit.. meh, idk - could change it to accept `sign_key: String` directly and make the callers do the `if let Some(..) = ..`?

Reply:
[Since `sign_settings.key` is `Option<String>`](https://github.com/pylbrecht/jj/blob/3edbfa34c327a0f7e16f95a1ae0aebe4381f39c6/lib/src/settings.rs#L90), wouldn't we eliminate the case, where the caller wants to explicitly set `sign_settings.key = None`?
Or is this case nonsense and `sign_settings.key` should never be set to `None` after initialization ([`SignSettings::from_settings()`](https://github.com/pylbrecht/jj/blob/3edbfa34c327a0f7e16f95a1ae0aebe4381f39c6/lib/src/settings.rs#L104))?

Reply:
@necauqua, I reverted the change as I honestly do not recall why I did it in the first place. 🤔 
Let me know if you have any objections.

Reply:
Well, I never loaded the entire context of this specific PR into my head, so I don't have any objections whatsoever
If it works, it works, and if it also happens to make sense, the better)